_posts/2019-01-08-data-driven-static-resume.adoc

```diff
@@ -0,0 +1,202 @@
+
+.From resume.asciidoc -- Template mixes Liquid with AsciiDoc, generates new AsciiDoc source
+[source]
+----
+include::assets/includes/resume.asciidoc[tags="employment"]
+----
+
+Because Liquid templates are usually a mix of at least two markup languages (Liquid markup itself plus your target format), they can get pretty ugly.
+Such is the nature of content templating.
+Nevertheless, we can generate such elegant code this way.
+
+Don't worry about the details of this code.
+Basically what it does is iterate (“loop”) through the `posts.paid` array in the data file, creating an entry for each item it comes across (objects describing my past paid gigs).
+As it churns, Liquid expresses the details of each object, mixing it in with some AsciiDoc markup you'll also recognize in the output just below.
+
+.From _build/pages/resume.adoc -- AsciiDoc source file output after parsing
+[source,asciidoc]
+----
+include::pages/resume.adoc[tags="example-snippet"]
+----
+
+The above snippet includes just the AsciiDoc generated for the first two entries, though the `_build/pages/resume.adoc` file is a complete AsciiDoc document source.
+
+[NOTE]
+Since this file is built and thus not tracked in the Git repository, to see the whole file you would have to clone and build the Codewriting repo, generating a copy into the `_build/pages/` directory.)
+
+[TIP]
+Because that final, prebuilt source is generated before this blog, the above snippet is directly sourced from the resulting file, accessed while this blog post was being rendered and ensuring it always accurately reflects the _current_ state of the underlying data.
+(See the <<sniptip,bonus explainer>> at the end of this post.)
+
+Finally, I render that AsciiDoc file into rich-text output.
+The PDF document is generated by the killer link:http://prawnpdf.org/api-docs/2.0/[*Prawn* PDF utility], after which the link:https://jekyllrb.com/[Jekyll static site generator] spits it out as HTML.
+Both of these processes invoke the link:https://github.com/asciidoctor/asciidoctor[*Asciidoctor*] rendering engine, as detailed below in <<the-build>>.
+
+== The Output
+
+Here are images from the relevant details of both rendered artifacts.
+
+.HTML version
+====
+image::assets/images/screenshot-detail_resume-employment-html.png[]
+====
+
+.PDF version
+====
+image::assets/images/screenshot-detail_resume-employment-pdf.png[]
+====
+
+[[the-build]]
+== The Build
+
+Now that we see what happened, let's explore how it was _scripted_.
+
+All of this activity is coordinated by link:https://www.ajyl.org/liquidoc[*LiquiDoc*], my free, open-source utility for combining AsciiDoc, YAML, and Liquid with Jekyll and other link:https://jamstack.org/[JAMstack services and utilities].
+LiquiDoc is a nascent API for the AJYL docstack, designed for building rich-text documents in many editions and formats, all from the same source, by invoking upstream tools (such as Asciidoctor, Liquid, and Jekyll processing engines).
```

Comment:
```suggestion
LiquiDoc is a nascent API for the AJYL docstack.
It is designed for building rich-text documents in many editions and formats, all from the same source, by invoking upstream tools (such as Asciidoctor, Liquid, and Jekyll processing engines).
```

This is a fab description, and it gets bogged down or convoluted by the long sentence.